### PR TITLE
zhen sgg 4

### PR DESCRIPTION
**Pull request by renztim**

### CODE REVIEW

File: source/Sgghost.h
Comment (emilydolson) [inline source/Sgghost.h:76]: For maintainability (since only the last couple lines of this for loop are different from the base class), it would probably be better to have this call the version of DistribResources in the base class and then make a new loop for the pool stuff. The more we can avoid duplicated code in a project like this the better. So the function would look like this:

```
void DistribResources(double resources) {
	Host::DistribResources(resources);
	for(size_t i=0; i < syms.size(); i++){
		double hostPool = syms[i]->ProcessPool();
		this->AddPool(hostPool);
	}
	if(syms.size()>0){this->DistribPool();}

} //end DistribResources
```

File: source/Sgghost.h
Comment (emilydolson) [inline source/Sgghost.h:94]: Looks like this function is getting run by any of the tests. Try to add at least one that runs it.

File: source/sggsym.h
Comment (emilydolson) [inline source/sggsym.h:24]: Could you rename these functions to GetDonation and SetDonation for consistency with the style of the rest of the code?

File: source/sggsym.h
Comment (emilydolson) [inline source/sggsym.h:29]: Could replace this if statement with a call to `Symbiont::mutate()` to reduce the amount of duplicate code (and insure that future changes to the base mutate function get reflected here)

File: source/sggsym.h
Comment (emilydolson) [inline source/sggsym.h:46]: This too

File: source/sggsym.h
Comment (emilydolson) [inline source/sggsym.h:98]: Might want to output the donation value too

**Review by emilydolson — CHANGES_REQUESTED** [review]
Looking good! Almost there! I made a few more in-line comments, but they're mostly pretty minor polishing things.

I still think it might be worth switching the SGG (for "Social Goods Game") prefix to PGG (for "Public Goods Game") because that is the more commonly used term. It should be a not-too-bad find and replace. What do you think, @anyaevostinar?

Is the other pull request (#59) something different than this one or can it get closed? If possible it would be great to keep all of this in a single pull request, for organization purposes (and so I can more easily see which code I've already looked at). Any commits that you put on the main branch of your fork should end up here and all the checks on this pull request will automatically update. Let me know if you have any questions!

One other question for @anyaevostinar about the contribution process for Symbulation: once I've approved this pull request, do you want to also review it before it gets merged?

File: source/ConfigSetup.h
Comment (anyaevostinar) [inline source/ConfigSetup.h:46]: This should go after the file_path and file_name since otherwise they will be lumped into the PGG group

File: source/Organism.h
Comment (anyaevostinar) [inline source/Organism.h:154]: These should each be the exact name of the method to make it easier to track down if/when it happens

File: source/PggHostPggSymUnitTest.test.cc
Comment (anyaevostinar) [inline source/PggHostPggSymUnitTest.test.cc:20]: It doesn't look like you have your own new methods for these, so I'm not sure why you need tests for them if you are just using the Host and Symbiont methods

File: source/WorldSetup.cc
Comment (anyaevostinar) [inline source/WorldSetup.cc:60]: This is a whole lot of repeated code, you could probably just have the emp::Ptr<Host> new org in an if statement instead from the looks of it.

**Review by anyaevostinar — CHANGES_REQUESTED** [review]
I've put in some comments focused on reducing repeated code/bloat

File: SymSettings.cfg
Comment (anyaevostinar) [inline SymSettings.cfg:49]: These are still changed from the default: "FILE_PATH" and "FILE_NAME"

File: SymSettings.cfg
Comment (anyaevostinar) [inline SymSettings.cfg:43]: These two still need to be changed back

File: source/Pggendtoend.test.cc
Comment (anyaevostinar) [inline source/Pggendtoend.test.cc:126]: It doesn't look like data files are actually being added to check against, so I don't think this test will do anything. I also think we are going to try to move away from this kind of test since it is difficult to maintain, so I would prefer you remove it.

File: source/Pgghost.h
Comment (anyaevostinar) [inline source/Pgghost.h:25]: You don't need this line in here since the Host constructor will do it, it's just repeated code.

File: source/Pgghost.test.cc
Comment (anyaevostinar) [inline source/Pgghost.test.cc:21]: Why do these need to be tested? Aren't they just using the Host's methods?

**Review by anyaevostinar — CHANGES_REQUESTED** [review]
Still some repeated code and tests that should get resolved

File: source/WorldSetup.cc
Comment (emilydolson) [reply source/WorldSetup.cc:60]: This looks like it will be easier to fix once #67 is resolved - should we hold off on fixing it until #67 is fixed?

Comment (anyaevostinar) [reply source/WorldSetup.cc:60]: I think it should be easy in this case since it looks like the only line that is different between these blocks is " emp::Ptr<PggHost> new_org;" versus "emp::Ptr<Host> new_org;" so the MakeNew wouldn't be needed anyway

Comment (emilydolson) [reply source/WorldSetup.cc:60]: Sorry if I'm being dumb here, but if he puts it in an if statement won't it go out of scope once the if statement ends?

File: source/Pgghost.h
Comment (emilydolson) [inline source/Pgghost.h:36]: I think I had a comment about this before, but it must have gotten lost. You can replace most of this function with a call to Host::DistribResources(resources); Then you just need to add a for loop after it to handle the pool stuff (which is the only difference between this function and the Host's version)

**Comment by emilydolson** [comment]
I cleaned up a couple minor differences that were making the git history messy and did one last readthrough of the changes. The only definite remaining issue that I see is the duplicated code in DistribResources, and possibly the thing in WorldSetup.cc

File: source/WorldSetup.cc
Comment (renztim) [reply source/WorldSetup.cc:60]: I tried to put the emp::Ptr new_org in an if statment before but it returns an error said that new_org was not decleared in the scope.

**Comment by emilydolson** [comment]
Great work! This addresses all of my comments, so I'm going to go ahead and approve. I think the only possible outstanding issue is that if statement.

**Review by emilydolson — APPROVED** [review]

**Review by anyaevostinar — APPROVED** [review]
Great job!

**Pull request merged**
